exploration/code-mode-introducer.md

```diff
@@ -0,0 +1,224 @@
+		<dd>#000</dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+It must be possible to reliably parse messages.
+
+Our current syntax features unquoted patterns for simple messages
+and unquoted code tokens with quoted patterns for complex messages.
+Determining whether a message will have code tokens requires some
+special character sequence, either part of the code itself or
+prepended to the message.
+This proposal examines the options for determining code mode.
+
+## Background
+
+_What context is helpful to understand this proposal?_
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+As a developer, I want to create messages with the minimal amount of special syntax.
+I don't want to have to type additional characters that add no value.
+I want the syntax to be logical and as consistent as possible.
+
+As a translator, I don't want to have to learn special syntax to support features such as declarations.
+
+As a user, I want my messages to be robust.
+Minor edits and changes should not result in syntax errors.
+
+As a user, I want to be able to see which messages are complex at a glance
+and to parse messages into their component parts visually as easily as possible.
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+Some of the options use a new sigil as part of the introducer.
+For various reasons, `#` has been used recently as a placeholder for this sigil.
+There are concerns that this character is not suitable, since it is used as a comment
+introducer in a number of formats. 
+See for example [#520](https://github.com/unicode-org/message-format-wg/issues/520).
+The actual sigil used needs to be an ASCII character in the reserved or private use
+set (with syntax adjustments if we use up a private-use one).
+Most of the options below have been changed to use `^`, using 
+Apple's experimental syntax as a model for sigil choice.
+
+It should be noted that an introducer sigil should be as rare as possible in normal text.
+This tends to run against common punctuation marks `&`, `%`, `!`, and `?`.
+
+```abnf
+reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
+private-start  = "^" / "&"
+```
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+We need to choose one of these (or another option not yet considered).
+Presentation at UTW did not produce any opinions.
+
+Based on the pro/cons below, I would suggest Option D is possibly the best option?
+
+## Alternatives Considered
+
+_What other solutions are available?_
+_How do they compare against the requirements?_
+_What other properties they have?_
+
+There are the following designs being considered:
+
+### Option A. Use Pattern Quotes for Messages (current design)
```

Comment:
We could also consider slight variations of Option A which aren't exactly the current `main` syntax. For example: `{# ... #}`, `{% ... %}`, or `{[ ... ]}`. They could help mitigate at least the `}}}}` problem.

Reply:
Good callout, although it's still a lot of closing syntax `}}%}`